Improve encoding section (#2735)
altair-viz/altair: doc/user_guide/encoding.rst

Here the y-axis is sorted reverse-alphabetically, while the color legend is
sorted in the specified order, beginning with ``'Morris'``.

Datum and Value
~~~~~~~~~~~~~~~

So far we always mapped an encoding channel to a column in our dataset. However, sometimes
it is also useful to map to a single constant value. In Altair, you can do this with

> * ``datum``, which describes a constant data value encoded via a scale
> * ``value``, which describes an encoded constant visual value
> 

Reviewer: ```suggestion
* ``datum``, which encodes a constant domain value via a scale using the same units as the underlying data
* ``value``, which encodes a constant visual value, using absolute units such as an exact position in pixels, the name or RGB value of a color, the name of shape,  etc

```

``datum`` is particularly useful for annotating a specific data value. 
For example, you can use it with a rule mark to highlight a 

> threshold value (e.g., 200 dollars stock price).

Reviewer: ```suggestion
threshold value (e.g., 300 dollars stock price).
```

.. altair-plot::
    
    import altair as alt
    from vega_datasets import data

    source = data.stocks()
    base = alt.Chart(source)
    lines = base.mark_line().encode(
        x="date:T",
        y="price:Q",
        color="symbol:N"
    )

>     rule = base.mark_rule(strokeDash=[2, 2]).encode(
>         y=alt.datum(300)
>     )
> 
>     lines + rule

Reviewer: ```suggestion
    rule = base.mark_rule(strokeDash=[2, 2]).encode(
        y=alt.datum(300)
    )

    lines + rule

If we instead used `alt.value` in this example, we would position the rule 300 pixels from the left of the chart border rather than at the 300 dollars position:

.. altair-plot::

    rule = base.mark_rule(strokeDash=[2, 2]).encode(
        y=alt.datum(300)
    )

    lines + rule
```
